File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int size = s.length();
        int arr[256] = {0};
        int left = 0, right = 0;
        int max_len = 0;

        while (right < size) {
            arr[s[right]]++;
            while (arr[s[right]] > 1) {
                arr[s[left]]--;
                left++;
            }
            max_len = max(max_len, right - left + 1);
            right++;
        }
        return max_len;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max_len = 0;
        int left  = 0, right = 0;
        int[] arr = new int[256];
        int res = 0;

        while(right < s.length()) {
            int curr = s.charAt(right);
            arr[curr]++;
            while(arr[curr] > 1) {
                arr[s.charAt(left)]--;
                left++;
            }
            res = Math.max(res, right - left + 1);
            right++;
        }
        return res;
    }
}